added Russian translation

StrikerRUS opened the pull request:
Based on comments from #60.

StrikerRUS committed:
added Russian translation

nabokovas commented on _i18n/ru/sidebar.md:
```diff
@@ -0,0 +1,24 @@
+Целью проекта QMK является разработка полностью настраиваемого, мощного и легкого в использовании программного обеспечения для любого проекта, будь то клавиатура или что-либо другое. Кроме этого, мы хотим помочь людям с любым уровнем опыта разработки программного обеспечения путем предоставления полезной, подбадривающей и искренней поддержки вместе с обратной связью.
```
What do you think about changing the word "искренней" for '...подбадривающей и любезной поддержки вместе с обратной связью'

nabokovas reviewed (changes requested):
The translation is good and only one word seemed to me inappropriate.

StrikerRUS committed:
address review comment

StrikerRUS commented:
@nabokovas Thanks a lot for your review! I fixed the wording in the latest commit.

lex111 commented on _data/strings.yml:
```diff
@@ -11,3 +11,11 @@ locales:
+    <<: *DEFAULT_EN
+    language_name: "Русский"
+    language: "Язык"
+    qmk_firmware: "QMK Прошивка"
```
```suggestion
    qmk_firmware: "Прошивка QMK"
```

lex111 replied:
Не пойму, это же ведь неправильно? QMK - название прошивки, следовательно единственный верный вариант - "Прошивка [название прошивки]" -> "Прошивка QMK", либо через дефис тогда - "QMK-прошивка", но не то. Может всё-таки поменяете, пожалуйста?

lex111 replied:
cc @StrikerRUS

StrikerRUS replied:
Я не против поменять. Но так уже написано в других местах этого проекта. Я не считаю это ошибкой, потому что это все - название продукта `Прошивка QMK` (обратите внимание на большую букву в слове `Прошивка`). Если бы это было `прошивка от фирмы QMK`, то да - ошибка. А так это просто разный перевод названия 🙂 . Поэтому предпочту однообразие в угоду более звучащему названию.

lex111 replied:
А где в других местах написано, в оригинале то есть? Если название продукта, то тогда переводить его ещё большая ошибка.
Я понял вашу мысль, но она очень не слишком очевидная (в том смысле, что это название проекта в целом), потому что в английском зачастую пишут заголовки, где слова идут с большой буквы. Конечно, вы не хуже вы меня это знаете, но не могу не привести пример с документацией - https://docs.qmk.fm/#/cli тут все заголовки в таком стиле. Поэтому я так ратую за это изменение, поскольку это частая ошибка при переводе :(

lex111 replied:
Вот даже в [немецком переводе](https://qmk.fm/de/) написано через дефис, что только подтверждает, что QMK - имя собственное, поэтому лучше изменить или тогда не переводить, не знаю, всяко лучше, чем сейчас.

StrikerRUS replied:
Нет, я не про заголовки, конечно же, вот, например, 
> QMK Firmware is a keyboard firmware based on the ...
https://github.com/qmk/qmk_firmware/tree/master/docs#what-is-qmk-firmware

Насчет того, что сам по себе перевод названия - это ошибка, пожалуй, не согласушь. Но оставить название чисто на английском языке не против. Только опять же: это надо сделать согласованно для всего проекта в целом. Поэтому предлагаю пока оставить так, а потом создать отдельный пуллреквест, где изменить всё разом.

lex111 replied:
> Насчет того, что сам по себе перевод названия - это ошибка, пожалуй, не согласушь.

Странно, конечно, потому что названия не переводят. Например, есть проект Microsoft Office, но вы ведь не будете писать его на русском как "Microsoft Офис"? Если посмотрите источники, никто так не делает, и это только один пример, здесь как я теперь понимаю, аналогичный случай.

StrikerRUS replied:
Ну про **никто** это как-то уж слишком! Почему-то же "Блокнот" в Windows перевели... Делают по-разному: и **не** переводят совсем, и переводят **частично**, и используют **транслитерацию**.

lex111 replied:
Ну хорошо, но вот представьте, что ваш любимый ЯП или фреймворк переводили, или использовали транслитерационное название. Согласен, ситуации разные, про "никто" было в контексте приведенного примера с Microsoft Office (как раз, кстати, наверное, чтобы избежать путаницы между офисом как физическим объектом и компьютерной программой, или как-то так; здесь похожее - либо само название прошивки (что тоже так), либо общее название проекта, куда входит прошивка.
Но всё это довольно странно, непривычно, но главное вряд ли уместно и вызывает недоумение (в данном случае закрадывается мысль, что это ошибка перевода ещё, словно просто заменили английское слово на русское, а в итоге не совсем ясное русифицированное название). Тем не менее, здесь ясно уже, что это название проекта, поэтому логично оставить его как есть (имена как раз уместно писать по-русски, хотя не лишним, например, указать в скобках оригинальное имя, это я забыл сделать).
В конце концов, само название проекта _QMK Firmware_, например, может измениться, тогда в будущем легче сделать операцию по поиску и замене. Этого в 99% не произойдет, но не знаю уже какой привести пример, чтобы показать, насколько это на самом деле важно.
Ладно, как говорится, хозяин - барин, просто хотел помочь в переводе, не сочтите, пожалуйста, за обиду.

StrikerRUS replied:
Так я же изначально был **НЕ** против полностью английского названия *QMK Firmware*! Проблема лишь в том, что `QMK Прошивка` **УЖЕ** находится в `master` в некоторых местах. Надо тогда будет тщательно прошерстить всё и сделать единообразно (*QMK Firmware*). Кто этим займется?

А вот, кстати, по поводу оригинальных имен в скобках после транслитерационных очень классная идея!

Раз уж мы бросаемся примерами, связанными с Майкрософт, хотел вот что показать. Так что, мне кажется, что переводят по-разному все и везде. Согласен, что это проблема.

![image](https://user-images.githubusercontent.com/25141164/66522254-a9e8bc80-eaf5-11e9-8837-0bde61596cc6.png)

> Ладно, как говорится, хозяин - барин, просто хотел помочь в переводе, не сочтите, пожалуйста, за обиду.

Да что вы! Какая обида? Рабочие моменты, так сказать. 🙂

nabokovas replied:
Если нужна помощь, то я могу пройтись по репозиторию и исправить русский перевод названия на QMK Firmware в других местах. Как вам идея?

StrikerRUS replied:
Отлично! Тогда я здесь возвращаю англ. название, а @nabokovas откатит то, что уже в `master`.

lex111 replied:
Значит я недопонял, просто я попробовал поискать вариант `QMK Прошивка` и не нашёл его нигде, тем более перевода документации ведь нет, а где ещё может быть такой перевод названия?
Насчет примера, да, тут всё не так просто, но им я просто хотел донести мысль, что как правило не переводят названия продуктов, брендов, компаний и т.п. Например, когда-то был Windows Media Center, хотя его тоже можно было перевести. Тут ещё такая неоднозначная ситуация, что QMK Firmware это одновременно и название проекта, и просто сама прошивка. Но тут да, верно указали, что второе слово не просто так написано с большой буквы, и это не имеет отношения к стилю написания английских заголовков, как я предполагал ранее. Но учитывая, что это всё-таки название проекта, то переводить лучше его не стоит, хотя бы из-за того, что трактуется как неверный перевод QMK firmware, подразумевая _прошивка QMK_.
Что ж, хорошо, что разобрались и пришли к общему и правильному решению :smiley:

lex111 commented on _i18n/ru/index.md:
```diff
@@ -0,0 +1,35 @@
+permalink: /ru/
+lang: "ru"
+---
+QMK - это прошивка для клавиатур, основанная на [прошивке tmk\_keyboard](http://github.com/tmk/tmk_keyboard) и включающая в себя много полезных изменений для Atmel AVR и ARM микроконтроллеров, а также поддерживающая продукты компаний [OLKB](https://olkb.com), [ErgoDox EZ](https://ergodox-ez.com/), и [Clueboard](https://clueboard.co/).
```
```suggestion
QMK - это прошивка для клавиатур, основанная на [прошивке tmk\_keyboard](http://github.com/tmk/tmk_keyboard), которая включает много полезных изменений для микроконтроллеров Atmel AVR и ARM, в частности для продуктов компаний [OLKB](https://olkb.com), [ErgoDox EZ](https://ergodox-ez.com/) и [Clueboard](https://clueboard.co/).
```

StrikerRUS replied:
Первое изменение `которая включает` меняет смысл предложения, относя вторую его часть к `tmk_keyboard`.

lex111 replied:
Точно, спасибо, пропустил точку.

```suggestion
QMK - это прошивка для клавиатур, основанная на [прошивке tmk\_keyboard](http://github.com/tmk/tmk_keyboard), и которая включает много полезных изменений для микроконтроллеров Atmel AVR и ARM, в частности для продуктов компаний [OLKB](https://olkb.com), [ErgoDox EZ](https://ergodox-ez.com/) и [Clueboard](https://clueboard.co/).
```

Но вообще здесь такая же ошибка как в первом изменении, потому что должно быть "микроконтроллеров Atmel AVR и ARM", можете, пожалуйста, это скорректировать? Ведь текущий вариант неправильный с точки зрения русского языка, разве нет?

StrikerRUS replied:
Тут, наверное, соглашусь, потому что здесь слово "микроконтроллер" уже не часть названия.

lex111 commented on _i18n/ru/index.md:
```diff
@@ -0,0 +1,35 @@
+
+## Разработчики
+
+QMK разработана и поддерживается Jack Humbert из OLKB вместе с соавторами из сообщества и, конечно же, [Hasu](https://github.com/tmk). Прошивки для OLKB разрабатываются [Jack Humbert](https://github.com/jackhumbert), для ErgoDox EZ - [ZSA Technology Labs](https://github.com/zsa), а для Clueboard - [Zach White](https://github.com/skullydazed).
```
```suggestion
QMK разработана и поддерживается Джеком Гумбертом из OLKB совместно с участниками сообщества, а также, конечно же, [Хасу](https://github.com/tmk). Прошивки для OLKB поддерживаются [Джеком Гумбертом](https://github.com/jackhumbert), для ErgoDox EZ - [ZSA Technology Labs](https://github.com/zsa), а для Clueboard - [Заком Уайтом](https://github.com/skullydazed).
```

lex111 commented on _i18n/ru/index.md:
```diff
@@ -0,0 +1,35 @@
+
+## Документация
+
+[Документация](https://docs.qmk.fm) расположена на [GitBook](https://www.gitbook.com/book/qmk/firmware/details) и [GitHub](https://github.com/qmk/qmk_firmware/tree/master/docs) (они синхронизированы). Вы можете предложить изменения, создав fork и [pull request](https://github.com/qmk/qmk_firmware/pulls), или кликнув по "suggest an edit" на любой странице.
```
According to https://github.com/web-standards-ru/dictionary/blob/master/git.md#словарь-терминов-git-и-github

```suggestion
[Документация](https://docs.qmk.fm) находится на [GitBook](https://www.gitbook.com/book/qmk/firmware/details) и [GitHub](https://github.com/qmk/qmk_firmware/tree/master/docs) (они синхронизированы). Вы можете предложить изменения, создав форк репозитория и открыть [пулреквест](https://github.com/qmk/qmk_firmware/pulls), либо кликнув по "suggest an edit" на любой странице.
```

lex111 commented on _i18n/ru/index.md:
```diff
@@ -0,0 +1,35 @@
+
+## Конфигуратор
+
+Используйте наш [онлайн конфигуратор](https://config.qmk.fm), чтобы с комфортом сконструировать файл раскладки, а затем скомпилировать и скачать прошивку прямо из своего браузера. На данный момент поддерживаются только Chrome и Firefox.
```
```suggestion
Воспользуйтесь [онлайн-конфигуратором](https://config.qmk.fm), чтобы прямо из браузера создать файл раскладки, а затем скомпилировать и скачать прошивку. На данный момент поддерживаются только Chrome и Firefox.
```

lex111 commented on _i18n/ru/sidebar.md:
```diff
@@ -0,0 +1,24 @@
+
+### Страницы
+
+* [Онлайн Конфигуратор](https://config.qmk.fm)
```
```suggestion
* [Онлайн-конфигуратор](https://config.qmk.fm)
```

lex111 commented on _i18n/ru/sidebar.md:
```diff
@@ -0,0 +1,24 @@
+### Страницы
+
+* [Онлайн Конфигуратор](https://config.qmk.fm)
+* [Недавние Изменения в QMK](/changes/)
```
```suggestion
* [Последние изменения в QMK](/changes/)
```

lex111 commented on _i18n/ru/sidebar.md:
```diff
@@ -0,0 +1,24 @@
+
+* [Онлайн Конфигуратор](https://config.qmk.fm)
+* [Недавние Изменения в QMK](/changes/)
+* [Все Поддерживаемые Клавиатуры](/keyboards/)
```
```suggestion
* [Список поддерживаемых клавиатур](/keyboards/)
```

lex111 commented on _i18n/ru/sidebar.md:
```diff
@@ -0,0 +1,24 @@
+* [Онлайн Конфигуратор](https://config.qmk.fm)
+* [Недавние Изменения в QMK](/changes/)
+* [Все Поддерживаемые Клавиатуры](/keyboards/)
+* [QMK для Вашей Клавиатуры](/powered/)
```
```suggestion
* [Про товарный знак QMK](/powered/)
```

lex111 commented on _i18n/ru/sidebar.md:
```diff
@@ -0,0 +1,24 @@
+* [Недавние Изменения в QMK](/changes/)
+* [Все Поддерживаемые Клавиатуры](/keyboards/)
+* [QMK для Вашей Клавиатуры](/powered/)
+* [Информация о Proton C](/proton-c/)
```
```suggestion
* [О микроконтроллере Proton C](/proton-c/)
```

lex111 commented on _i18n/ru/sidebar.md:
```diff
@@ -0,0 +1,24 @@
+* [Все Поддерживаемые Клавиатуры](/keyboards/)
+* [QMK для Вашей Клавиатуры](/powered/)
+* [Информация о Proton C](/proton-c/)
+* [Информация о Лицензии](/license/)
```
```suggestion
* [Информация о лицензии](/license/)
```

lex111 commented on _i18n/ru/sidebar.md:
```diff
@@ -0,0 +1,24 @@
+* [QMK для Вашей Клавиатуры](/powered/)
+* [Информация о Proton C](/proton-c/)
+* [Информация о Лицензии](/license/)
+* [Нормы Поведения](/coc/)
```
```suggestion
* [Нормы поведения](/coc/)
```

lex111 commented on _i18n/ru/sidebar.md:
```diff
@@ -0,0 +1,24 @@
+* [Информация о Proton C](/proton-c/)
+* [Информация о Лицензии](/license/)
+* [Нормы Поведения](/coc/)
+* [Как Поддержать QMK](/support/)
```
```suggestion
* [Как поддержать QMK](/support/)
```

lex111 commented on _i18n/ru/index.md:
```diff
@@ -0,0 +1,35 @@
+* [Georgi](https://qmk.fm/keyboards/georgi/)
+* [Atreus](https://github.com/qmk/qmk_firmware/blob/master/keyboards/atreus/)
+
+Эти клавиатуры используют QMK в качестве своей официальной и установленной по умолчанию прошивки. В списке не хватает вашего проекта? [Сообщите нам!](https://github.com/qmk/qmk.fm/issues/new) 
```
```suggestion
Эти клавиатуры используют QMK в качестве официальной и установленной по умолчанию прошивки. В списке нет вашего проекта? [Сообщите нам!](https://github.com/qmk/qmk.fm/issues/new) 
```

lex111 replied:
Вкусовщина разве? Притяжательное местоимение точно можно убрать, намного лучше читается. Ну, как знаете.

lex111 commented on _data/strings.yml:
```diff
@@ -11,3 +11,11 @@ locales:
+    language: "Язык"
+    qmk_firmware: "QMK Прошивка"
+    made_possible: "Данный проект поддерживается <a href='https://github.com/qmk'>коллабораторами QMK</a> и такими же соавторами, как ты!"
+    hosted_on: "Хостинг предоставлен GitHub Pages &mdash; Тема создана <a href='https://github.com/orderedlist'>orderedlist</a>"
```
```suggestion
    hosted_on: "Хостинг от GitHub Pages, тема &mdash; <a href='https://github.com/orderedlist'>orderedlist</a>"
```

StrikerRUS replied:
`&mdash;` here is used as a delimiter for info about hosting and theme. It is not a part of a sentence about the theme. Please see eng sources.

lex111 replied:
Я конечно же смотрел оригинал, просто хотелось избежать переноса строки. Можно тогда упросить до такого, смысл остаётся такой же, да и нет "непонятных" слов вроде "хостинг".

```
Размещено на GitHub Pages — Тема от orderedlis
```

Как считаете?

lex111 replied:
cc @StrikerRUS

StrikerRUS replied:
Соглашусь!

lex111 commented on _data/strings.yml:
```diff
@@ -11,3 +11,11 @@ locales:
+    language_name: "Русский"
+    language: "Язык"
+    qmk_firmware: "QMK Прошивка"
+    made_possible: "Данный проект поддерживается <a href='https://github.com/qmk'>коллабораторами QMK</a> и такими же соавторами, как ты!"
```
```suggestion
    made_possible: "Данный проект поддерживается <a href='https://github.com/qmk'>коллабораторами QMK</a> и такими участниками, как вы!"
```

lex111 commented on _i18n/ru/changes.md:
```diff
@@ -0,0 +1,18 @@
+---
+title: "Все недавние изменения в QMK"
```
```suggestion
title: "Все последние изменения в QMK"
```

lex111 reviewed (commented):
Suggestions for improving translation, it is advisable to accept them all.

StrikerRUS committed:
addressed review comments

StrikerRUS commented:
@lex111 Thanks a lot for your detailed review! I've addressed some comments from it in the latest commit. All others are a matter of taste. And I'd prefer to leave my variants at least with the aim to keep it consistent with other (already merged) my translations across the project (not only in this particular repo). Feel free to propose another separate PR to paraphrase names (`Прошивка QMK` -> `QMK Прошивка`)!

StrikerRUS committed:
more review comments addressed

StrikerRUS committed:
removed excess comma

StrikerRUS committed:
do not translate firmware word in name

StrikerRUS committed:
transliterate names keeping original ones in brackets

lex111 reviewed (approved):
LGFM :+1:

StrikerRUS commented:
@jackhumbert Can we merge?

Shehanka commented:
Can I translate this to Sinhala

StrikerRUS commented:
@jackhumbert Any updates?

raxraj commented:
The translations are as perfect as it can be. Good Work

raghavdhingra commented:
Looks good

kuchosauronad0 commented on _i18n/ru/changes.md:
```diff
@@ -0,0 +1,18 @@
+permalink: /ru/changes/
+---
+
+{% for change in site.changes %}
```
```suggestion
{% for change in site.changes reversed %}
```
This way the most recent change is at the top of the page. See #78

StrikerRUS replied:
@kuchosauronad0 Thanks for letting know! Will commit this after merging the upstream PR.

noroadsleft replied:
I just merged PR 78, this can go in now.

StrikerRUS replied:
@noroadsleft Thanks, done!

StrikerRUS committed:
Update _i18n/ru/changes.md

Co-Authored-By: kuchosauronad0 <22005492+kuchosauronad0@users.noreply.github.com>

noroadsleft reviewed (approved):
Looks good, thanks!
